algebird-test/src/test/scala/com/twitter/algebird/MomentsLaws.scala

```diff
@@ -1,24 +1,48 @@
 package com.twitter.algebird
 
 import com.twitter.algebird.BaseProperties._
-import com.twitter.algebird.scalacheck.arbitrary._
 import org.scalatest.matchers.should.Matchers
 import org.scalatest.wordspec.AnyWordSpec
+import org.scalacheck.{Arbitrary, Gen}
 
 class MomentsLaws extends CheckProperties {
   val EPS = 1e-10
 
+  implicit val equiv: Equiv[Moments] =
+    Equiv.fromFunction { (ml, mr) =>
+      (ml.m0 == mr.m0) &&
+      approxEq(EPS)(ml.m1, mr.m1) &&
+      approxEq(EPS)(ml.m2, mr.m2) &&
+      approxEq(EPS)(ml.m3, mr.m3) &&
+      approxEq(EPS)(ml.m4, mr.m4)
+    }
+
+  def opBasedGen[A: Numeric](genA: Gen[A]): Gen[Moments] = {
+    val init: Gen[Moments] = genA.map(Moments(_))
+
+    val recur = Gen.lzy(opBasedGen[A](genA))
+    val pair = Gen.zip(recur, recur)
+
+    import Operators.Ops
+
+    Gen.frequency(
+      (10, init),
+      (1, pair.map { case (a, b) => a + b })
+    )
+  }
+
   property("Moments Group laws") {
-    implicit val equiv: Equiv[Moments] =
-      Equiv.fromFunction { (ml, mr) =>
-        (ml.m0 == mr.m0) &&
-        approxEq(EPS)(ml.m1, mr.m1) &&
-        approxEq(EPS)(ml.m2, mr.m2) &&
-        approxEq(EPS)(ml.m3, mr.m3) &&
-        approxEq(EPS)(ml.m4, mr.m4)
-      }
+    import com.twitter.algebird.scalacheck.arbitrary.momentsArb
+    implicit val group: Group[Moments] = MomentsGroup
     groupLaws[Moments]
```

Comment:
these laws continue to (usually) pass when we use the old generator. If you use a generator that includes generating pairs and subtracting them it almost always fails.